lms/services/group_info.py

```diff
@@ -36,9 +36,15 @@ def upsert(self, authority_provided_id, consumer_key, params):
 
         :param authority_provided_id: the ``authority_provided_id`` of the
             ``GroupInfo`` to create or update
+
         :param consumer_key: the ``GroupInfo.consumer_key`` value to set
+
         :param params: the other ``GroupInfo`` columns to set
         :type params: dict
+
+        :param type_: the type of the group, e.g. "course_group" or "section_group"
+        :type type_: str
+
         """
```

Comment:
Maybe go the other way and remove the types?

```suggestion
        :param authority_provided_id: the ``authority_provided_id`` of the
            ``GroupInfo`` to create or update
        :param consumer_key: the ``GroupInfo.consumer_key`` value to set
        :param params: the other ``GroupInfo`` columns to set
        :param type_: the type of the group, e.g. "course_group" or "section_group"
        """
```

Reply:
I've simplified this docstring: https://github.com/hypothesis/lms/pull/1717/commits/856ec13e8112002fcbfd86b90cd50f66af5eb1a8

I do think the `:type`'s on the `h_group` and `params` arguments are adding something so I left those in, but the one on `consumer_key` was just duplicating info from the `GroupInfo` model so I removed that one